src/middlewares/validateBody/index.ts

```diff
@@ -0,0 +1,41 @@
+import { Request, Response, NextFunction } from 'express';
+import Ajv, { JSONSchemaType } from 'ajv';
+
+const validateSchema = (data: unknown, schema: JSONSchemaType<unknown>) => {
+  const ajv = new Ajv();
+  const validate = ajv.compile(schema);
+
+  if (validate(data)) return true;
+  return false;
+};
+
+/**
+ * High order function that returns a middleware validating request body
+ * against a predefined schema. If request body is invalid, throws an error.
+ * Can be used for validating request bodies.
+ * @param schema The predefined schema
+ * @returns The middleware
+ */
+export const validateReqBody = (schema: JSONSchemaType<unknown>) => {
+  return (req: Request, res: Response, next: NextFunction) => {
+    if (validateSchema(req.body, schema)) {
+      next();
+    } else {
+      throw new Error('Invalid body');
+    }
+  };
+};
+
+/**
+ * Function that validate body against a predefined schema.
+ * If request body is invalid, throws an error.
+ * Can be used for validating socket bodies.
+ * @param schema The predefined schema
+ * @returns The middleware
```

Comment:
This function doesn't return the middleware, this line should be removed.
Also `@param data` shoud be added

Reply:
Thank you for your feedback. Everything is resolved and made the code documentation more consice.